model-optimizer/mo/graph/port.py

```diff
@@ -97,12 +97,12 @@ def _set_shape(self, shape):
         else:
             if self.type == 'in':
                 assert self.node.in_node(self.idx, control_flow=self.control_flow).value is None
-                self.node.in_node(self.idx, control_flow=self.control_flow).shape = int64_array(shape)
+                self.node.in_node(self.idx, control_flow=self.control_flow).shape = shape_array(shape)
             else:
                 data_node = self.node.out_node(self.idx, control_flow=self.control_flow)
                 assert data_node.value is None or \
-                       np.array_equal(data_node.soft_get('force_shape', data_node.shape), int64_array(shape))
-                self.node.out_node(self.idx, control_flow=self.control_flow).shape = int64_array(shape)
+                       np.ma.allequal(data_node.soft_get('force_shape', data_node.shape), shape_array(shape))
```

Comment:
np.ma.allequal -> compare_shape or `strict_compare` when `value` is defined.

Reply:
Changed.